src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector256.cs

```diff
@@ -517,77 +498,77 @@ public static unsafe Vector256<float> ConvertToSingle(Vector256<uint> vector)
+        public static Vector256<uint> ConvertToUInt32(Vector256<float> vector)
         {
-            Unsafe.SkipInit(out Vector256<uint> result);
-
-            for (int i = 0; i < Vector256<uint>.Count; i++)
-            {
-                uint value = (uint)vector.GetElementUnsafe(i);
-                result.SetElementUnsafe(i, value);
-            }
-
-            return result;
+            return Create(
+                Vector128.ConvertToUInt32(vector.GetLower()),
+                Vector128.ConvertToUInt32(vector.GetUpper())
+            );
         }
 
         /// <summary>Converts a <see cref="Vector256{Double}" /> to a <see cref="Vector256{UInt64}" />.</summary>
         /// <param name="vector">The vector to convert.</param>
         /// <returns>The converted vector.</returns>
-        [CLSCompliant(false)]
         [Intrinsic]
-        public static unsafe Vector256<ulong> ConvertToUInt64(Vector256<double> vector)
+        [CLSCompliant(false)]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector256<ulong> ConvertToUInt64(Vector256<double> vector)
         {
-            Unsafe.SkipInit(out Vector256<ulong> result);
-
-            for (int i = 0; i < Vector256<ulong>.Count; i++)
-            {
-                ulong value = (ulong)vector.GetElementUnsafe(i);
-                result.SetElementUnsafe(i, value);
-            }
-
-            return result;
+            return Create(
+                Vector128.ConvertToUInt64(vector.GetLower()),
+                Vector128.ConvertToUInt64(vector.GetUpper())
+            );
         }
 
         /// <summary>Copies a <see cref="Vector256{T}" /> to a given array.</summary>
         /// <typeparam name="T">The type of the elements in the vector.</typeparam>
         /// <param name="vector">The vector to be copied.</param>
         /// <param name="destination">The array to which <paramref name="vector" /> is copied.</param>
-        /// <exception cref="NullReferenceException"><paramref name="destination" /> is <c>null</c>.</exception>
         /// <exception cref="ArgumentException">The length of <paramref name="destination" /> is less than <see cref="Vector256{T}.Count" />.</exception>
+        /// <exception cref="NotSupportedException">The type of <paramref name="vector" /> and <paramref name="destination" /> (<typeparamref name="T" />) is not supported.</exception>
+        /// <exception cref="NullReferenceException"><paramref name="destination" /> is <c>null</c>.</exception>
         public static void CopyTo<T>(this Vector256<T> vector, T[] destination)
-            where T : struct => vector.CopyTo(destination, startIndex: 0);
+            where T : struct
+        {
+            ThrowHelper.ThrowForUnsupportedIntrinsicsVector256BaseType<T>();
+
+            if (destination is null)
+            {
+                ThrowHelper.ThrowNullReferenceException();
+            }
+
+            if (destination.Length < Vector256<T>.Count)
```

Comment:
Why is the above if block throwing a NullReferenceException necessary if here we're dereferencing destination anyway?

Reply:
Likely just a copy/paste thing that's been propagated over time. I can simplify as part of fixing the CI failures.